beacon-chain/das/availability_columns.go

```diff
@@ -112,23 +121,33 @@ func (s *LazilyPersistentStoreColumn) IsDataAvailable(
 	// ignore their response and decrease their peer score.
 	roDataColumns, err := entry.filter(blockRoot, blockCommitments)
 	if err != nil {
-		return errors.Wrap(err, "incomplete DataColumnSidecar batch")
+		return errors.Wrap(err, "filter")
+	}
+
+	// https://github.com/ethereum/consensus-specs/blob/dev/specs/fulu/p2p-interface.md#datacolumnsidecarsbyrange-v1
+	verifier := s.newDataColumnsVerifier(roDataColumns, verification.ByRangeRequestDataColumnSidecarRequirements)
+
+	if err := verifier.Valid(); err != nil {
+		return errors.Wrap(err, "valid")
 	}
 
-	// Create verified RO data columns from RO data columns.
-	verifiedRODataColumns := make([]blocks.VerifiedRODataColumn, 0, len(roDataColumns))
+	if err := verifier.SidecarInclusionProven(); err != nil {
+		return errors.Wrap(err, "sidecar inclusion proven")
+	}
 
-	for _, roDataColumn := range roDataColumns {
-		verifiedRODataColumn := blocks.NewVerifiedRODataColumn(roDataColumn)
-		verifiedRODataColumns = append(verifiedRODataColumns, verifiedRODataColumn)
+	if err := verifier.SidecarKzgProofVerified(); err != nil {
+		return errors.Wrap(err, "sidecar KZG proof verified")
+	}
```

Comment:
Why do we need to run `Valid`, `SidecarInclusionProven`, `SidecarKzgProofVerified` as separate steps - couldn't they just run under the hood when you call `VerifiedRODataColumns`?

Reply:
You mean just moving this block:

```go
	if err := verifier.Valid(); err != nil {
		return errors.Wrap(err, "valid")
	}

	if err := verifier.SidecarInclusionProven(); err != nil {
		return errors.Wrap(err, "sidecar inclusion proven")
	}

	if err := verifier.SidecarKzgProofVerified(); err != nil {
		return errors.Wrap(err, "sidecar KZG proof verified")
	}
```

into `verifier.VerifiedRODataColumns`
?

Reply:
yes